R/importance.R

```diff
@@ -36,131 +39,243 @@
 #'  the risk with the permuted/removed X divided by observed risk with all X; or
 #'  \code{"difference"}, which returns the difference between the risk with the
 #'  permuted/removed X and the observed risk.
+#' @param covariate_groups Optional named list covariate groups which will
+#'  invoke variable importance evaluation at the group-level, by
+#'  removing/permuting all covariates in the same group together. If covariates
+#'  in the task are not specified in the list of groups, then those covariates
+#'  will be added as additional single-covariate groups.
+#' @param B Optional integer specifying the number of permutations to average
+#'  the permuted risk over when \code{type = "permute"}.
+#' @param cores Number of cores for parallelizing the \code{B} permutations.
 #'
 #' @importFrom data.table data.table
+#' @importFrom parallel mclapply detectCores
 #'
 #' @return A \code{data.table} of variable importance for each covariate.
 #'
 #' @name importance
 #' @keywords variable importance
 #'
 #' @export
-importance <- function(fit, loss = NULL, fold_number = "validation",
+importance <- function(fit, eval_fun = NULL, fold_number = "validation",
                        type = c("remove", "permute"),
-                       importance_metric = c("ratio", "difference")) {
+                       importance_metric = c("ratio", "difference"),
+                       covariate_groups = NULL, B = 1L,
+                       cores = parallel::detectCores() - 1) {
```

Comment:
let's remove this option and instead have the user set the cores use via a `future::plan()` specification. as is, this could lead to clashes between `parallel` and `future` if users have already set a `future::plan()` as we recommend